EXT_WAVEFRONT.md

```diff
@@ -0,0 +1,122 @@
+Domain-Specific Naming Conventions for Coherent Wavefront Propagation Codes
+===========================================================================
+
+openPMD extension name: `WAVEFRONT`
+
+
+Introduction
+------------
+
+This extension is specifically designed for the domain of coherent wavefront propagation codes.
+
+Additional attributes on `series`
+---------------------------------
+On the `series` object, set the following attributes:
+
+  - `beamline`
+    - type: *(string)*
+    - scope: *optional*
+    - description: The string representation of the optical beamline.
+
+  - `temporal domain`
+    - type: *(string)*
+    - scope: *required*
+    - description: Indicates whether the data represents a field in time or
+      frequency (energy) domain.
+    - allowed values:
+      - `time`: The field is given for the time domain.
+      - `frequency`: The field is given for the frequency (energy) domain.
+
+  - `spatial domain`
+    - type: *(string)*
+    - scope: *required*
+    - description: Indicates whether the data represents a field in cartesian
+      (r) or reciprocal (k) space.
+    - allowed values:
+      - `r`: The field is given in cartesian space.
+      - `k`: The field is given in reciprocmal space.
+
+Additional records
+------------------
+- `z coordinate`
+    -type: *(floatX)*
+    -description: The z coordinate with respect to the beamline origin.
+    - `unitDimension = (1., 0., 0., 0., 0., 0., 0.)` (m)
+    - scope: *required*
+
+- `photon energy`
+    - type: *(floatX)* or *(intX)* or *(uintX)*
```

Comment:
On which occasion the energy has `int` type?

Reply:
this is just for convenience, so photon_energy=4000 (4 keV) would not raise an error.